x-pack/plugin/rollup/src/internalClusterTest/java/org/elasticsearch/xpack/downsample/DownsampleClusterDisruptionIT.java

```diff
@@ -0,0 +1,231 @@
+        mapping.endObject().endObject().endObject();
+        assertAcked(indicesAdmin().prepareCreate(sourceIndex).setSettings(settings.build()).setMapping(mapping).get());
+    }
+
+    public void testRollupIndexWithDataNodeDisruptions() throws IOException, InterruptedException {
+        final List<String> masterNodes = internalCluster().startMasterOnlyNodes(3);
+        final List<String> dataNodes = internalCluster().startDataOnlyNodes(3);
+        final String[] allNodes = internalCluster().getNodeNames();
+        ensureStableCluster(internalCluster().size());
+        ensureGreen();
+
+        assertEquals(masterNodes.size() + dataNodes.size(), Arrays.stream(allNodes).count());
+        assertEquals(Arrays.stream(allNodes).count(), internalCluster().size());
+        assertFalse(dataNodes.contains(masterNodes));
+        assertFalse(masterNodes.contains(dataNodes));
+
+        final String sourceIndex = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        final String rollupIndex = randomAlphaOfLength(11).toLowerCase(Locale.ROOT);
+        long startTime = LocalDateTime.parse("2020-09-09T18:00:00").atZone(ZoneId.of("UTC")).toInstant().toEpochMilli();
+        // NOTE: we need replicas otherwise we might restart the node with the only source index shard and the index would not be available
+        setup(sourceIndex, 1, 3, startTime);
+        final DownsampleConfig config = new DownsampleConfig(randomInterval(), TIMEOUT);
+        final DownsampleActionSingleNodeTests.SourceSupplier sourceSupplier = () -> {
+            final String ts = randomDateForInterval(config.getInterval(), startTime);
+            double labelDoubleValue = DATE_FORMATTER.parseMillis(ts);
+            final List<String> dimensionValues = new ArrayList<>(5);
+            for (int j = 0; j < randomIntBetween(1, 5); j++) {
+                dimensionValues.add(randomAlphaOfLength(6));
+            }
+            return XContentFactory.jsonBuilder()
+                .startObject()
+                .field(FIELD_TIMESTAMP, ts)
+                .field(FIELD_DIMENSION_1, randomFrom(dimensionValues))
+                .field(FIELD_DIMENSION_2, randomIntBetween(1, 10))
+                .field(FIELD_METRIC_LABEL_DOUBLE, labelDoubleValue)
+                .endObject();
+        };
+        bulkIndex(sourceIndex, sourceSupplier, 9_000);
+        prepareSourceIndex(sourceIndex);
+        CountDownLatch disruptionStart = new CountDownLatch(1);
+        CountDownLatch disruptionEnd = new CountDownLatch(1);
+
+        new Thread(() -> {
+            try {
+                ensureStableCluster(internalCluster().size());
+                disruptionStart.countDown();
+                for (int i = 0; i < 10; i++) {
+                    final String restartingNode = randomFrom(dataNodes);
+                    logger.info("Restarting random data node [" + restartingNode + "]");
+                    try {
+                        internalCluster().restartNode(restartingNode);
```

Comment:
This makes the test fail if the node is not connected...because for instance it was restarted before and still joining?